### DISCUSSION TITLE
请教如何在 web-lite 版本，通过浏览器 URL query 参数，打开指定 uri 的文件？

### DISCUSSION THREAD

**Discussion by fankangsong** [opensumi/core#2294] [Q&A]
例如，通过 `http://127.0.0.1/?uri=/github/opensumi/core/README.md` 访问，自动打开 README.md 文件，并可以同步文件树状态。

**Comment by erha19** [comment]
@fankangsong 基于 web-lite 项目的案例，要实现这个需求，首先需要先了解一下 GitHub 的 Open API 能力https://docs.github.com/en/graphql ，然后示例代码内有针对 Query 获取参数的案例，你可以对照看一下实现，如：

这里定义了文件树请求的入口：
https://github.com/opensumi/ide-startup-lite/blob/d2aa96cef1bf96ad2477c9b590fd18661993ee72/src/app.tsx#L64

这里定义了通过 `BrowserFS` 在 LocalStorage 下模拟了文件系统（如果想每次打开都走网络请求，需要清理一下 LocalStorage，否则下次进入会默认走缓存）
https://github.com/opensumi/ide-startup-lite/blob/d2aa96cef1bf96ad2477c9b590fd18661993ee72/src/render-app.tsx#L31

读取文件是通过 HttpFileService 读取：
https://github.com/opensumi/ide-startup-lite/blob/00eb25aefd4c72217978fcf5f6921c900566ed52/web-lite/file-provider/http-file.service.ts#L132

可以自己尝试理解后修改一下试试

**Reply by fankangsong** [reply]
感谢回复。

BrowserFS 和 file-provider 部分的代码我阅读过了，前者是在浏览器虚拟文件系统，HttpFileService 是 BrowserFsProvider  的被调方，以及 github api 我也没有疑问。

在这个场景里不知道如何解决的问题是，通过 url query 参数读取要打开的文件路径后，opensumi 的编辑器要如何打开这个文件，有哪个模块提供类似命令式 api 可以调用吗，例如：

```typescript
await editor.open(`file://github/opensumi/core/README.md`)
```

**Reply by Ricbet** [reply]
有个内置的命令你可以试一下
```typescript
this.commandService.executeCommand(EDITOR_COMMANDS.OPEN_RESOURCE.id, ${URL})
```

**Reply by fankangsong** [reply]
> 有个内置的命令你可以试一下
> 
> ```ts
> this.commandService.executeCommand(EDITOR_COMMANDS.OPEN_RESOURCE.id, ${URL})
> ```

thx，这个方法可以。
